### PR TITLE
Added pending flag for operation execution

### PR DESCRIPTION
**Pull request by andreiblt1304**

### CODE REVIEW

File: esdt-safe/src/from_sovereign/transfer_tokens.rs [hunk 1]
```diff
@@ -290,27 +287,28 @@ pub trait TransferTokensModule:
...
+    }
+
+    fn lock_operation_hash(&self, operation_hash: &ManagedBuffer, hash_of_hashes: &ManagedBuffer) {
+        let header_verifier_address = self.header_verifier_address().get();
+
+        self.tx()
+            .to(header_verifier_address)
+            .typed(header_verifier_proxy::HeaderverifierProxy)
+            .lock_operation_hash(hash_of_hashes, operation_hash)
+            .returns(ReturnsResult)
```
Comment (axenteoctavian) [inline esdt-safe/src/from_sovereign/transfer_tokens.rs:310]: do you really need this?

File: esdt-safe/src/from_sovereign/transfer_tokens.rs [hunk 2]
```diff
@@ -321,16 +319,9 @@ pub trait TransferTokensModule:
         esdt_info_mapper.get().token_nonce
     }
 
-    #[storage_mapper("pendingHashes")]
-    fn pending_hashes(&self, hash_of_hashes: &ManagedBuffer) -> UnorderedSetMapper<ManagedBuffer>;
+    #[storage_mapper("pendingFlag")]
+    fn pending_flag(&self, hash_of_hashes: &ManagedBuffer) -> SingleValueMapper<bool>;
```
Comment (axenteoctavian) [inline esdt-safe/src/from_sovereign/transfer_tokens.rs:323]: this storage mapper is not used

File: header-verifier/src/lib.rs [hunk 3]
```diff
@@ -56,19 +55,42 @@ pub trait Headerverifier: bls_signature::BlsSignatureModule {
 
     #[endpoint(removeExecutedHash)]
     fn remove_executed_hash(&self, hash_of_hashes: &ManagedBuffer, operation_hash: &ManagedBuffer) {
+        self.require_caller_esdt_safe();
```
Comment (axenteoctavian) [inline header-verifier/src/lib.rs:58]:
```suggestion
        self.require_caller_esdt_safe();
        
```

File: header-verifier/src/lib.rs [hunk 4]
```diff
@@ -56,19 +55,42 @@ pub trait Headerverifier: bls_signature::BlsSignatureModule {
...
+        self.pending_hash(hash_of_hashes, operation_hash).clear();
+    }
+
+    #[endpoint(lockOperationHash)]
+    fn lock_operation_hash(&self, hash_of_hashes: ManagedBuffer, operation_hash: ManagedBuffer) {
+        self.require_caller_esdt_safe();
```
Comment (axenteoctavian) [inline header-verifier/src/lib.rs:64]:
```suggestion
        self.require_caller_esdt_safe();
        
```

File: esdt-safe/src/from_sovereign/transfer_tokens.rs [hunk 5]
```diff
@@ -290,27 +287,28 @@ pub trait TransferTokensModule:
     }
 
     // use pending_operations as param
```
Comment (axenteoctavian) [inline esdt-safe/src/from_sovereign/transfer_tokens.rs:289]: what's this comment about?

File: header-verifier/src/lib.rs [hunk 6]
```diff
@@ -56,19 +55,42 @@ pub trait Headerverifier: bls_signature::BlsSignatureModule {
...
+        let pending_hashes_mapper = self.pending_hash(&hash_of_hashes, &operation_hash);
```
Comment (axenteoctavian) [inline header-verifier/src/lib.rs:65]:
```suggestion
        let pending_hash_lock = self.pending_hash(&hash_of_hashes, &operation_hash);
```

File: esdt-safe/src/from_sovereign/transfer_tokens.rs (hunk 5 above)
Comment (andreiblt1304) [reply esdt-safe/src/from_sovereign/transfer_tokens.rs:289]: i forgot, i will remove it

File: header-verifier/tests/header_verifier_blackbox_test.rs [hunk 7]
```diff
@@ -350,7 +376,71 @@ fn test_remove_all_executed_hashes() {
         .whitebox(header_verifier::contract_obj, |sc| {
             let hash_of_hashes: ManagedBuffer<DebugApi> =
                 ManagedBuffer::from(operation.bridge_operation_hash.to_vec());
-            assert!(sc.pending_hashes(&hash_of_hashes).is_empty());
+            // assert!(sc.pending_hash(&hash_of_hashes).is_empty());
```
Comment (axenteoctavian) [inline header-verifier/tests/header_verifier_blackbox_test.rs:379]: why is this commented?

Comment (andreiblt1304) [reply header-verifier/tests/header_verifier_blackbox_test.rs:379]: forgot about it

**Review by axenteoctavian — APPROVED** [review]

**Review by alyn509 — APPROVED** [review]

**Pull request merged**